freqtrade/leverage/liquidation_price.py

```diff
@@ -41,147 +53,90 @@ def exception(
         :param exchange: Name of the exchange
-        :param margin_mode: one-way or hedge
         :param trading_mode: spot, margin, futures
         :param collateral: cross, isolated
     """
-    if not margin_mode:
-        raise OperationalException(
-            f"{exchange} does not support {collateral.value} {trading_mode.value} trading ")
 
     raise OperationalException(
-        f"{exchange} does not support {collateral.value} {margin_mode.value} Mode {trading_mode.value} trading ")
+        f"{exchange} does not support {collateral.value} Mode {trading_mode.value} trading ")
 
 
 def binance(
     open_rate: float,
     is_short: bool,
     leverage: float,
-    margin_mode: MarginMode,
     trading_mode: TradingMode,
     collateral: Collateral,
-    **kwargs
+    wallet_balance: float,
+    maintenance_margin_ex_1: float,
+    unrealized_pnl_ex_1: float,
+    maintenance_amount: float,
+    position_1: float,
+    entry_price_1: float,
+    maintenance_margin_rate: float,
 ):
     r"""
         Calculates the liquidation price on Binance
         :param open_rate: open_rate
         :param is_short: true or false
         :param leverage: leverage in float
-        :param margin_mode: one-way or hedge
         :param trading_mode: spot, margin, futures
         :param collateral: cross, isolated
 
-        :param \**kwargs:
-            See below
-
-        :Keyword Arguments:
-            * *wallet_balance* (``float``) --
-              Wallet Balance is crossWalletBalance in Cross-Margin Mode
-              Wallet Balance is isolatedWalletBalance in Isolated Margin Mode
-
-            * *maintenance_margin_ex_1* (``float``) --
-              Maintenance Margin of all other contracts, excluding Contract 1.
-              If it is an isolated margin mode, then TMM=0
-
-            * *unrealized_pnl_ex_1* (``float``) --
-              Unrealized PNL of all other contracts, excluding Contract 1.
-              If it is an isolated margin mode, then UPNL=0
-
-            * *maintenance_amount_both* (``float``) --
-              Maintenance Amount of BOTH position (one-way mode)
-
-            * *maintenance_amount_long* (``float``) --
-              Maintenance Amount of LONG position (hedge mode)
+        :param wallet_balance: Wallet Balance is crossWalletBalance in Cross-Margin Mode.
+          Wallet Balance is isolatedWalletBalance in Isolated Margin Mode
 
-            * *maintenance_amount_short* (``float``) --
-              Maintenance Amount of SHORT position (hedge mode)
+        :param maintenance_margin_ex_1: Maintenance Margin of all other contracts, excluding Contract 1.
+          If it is an isolated margin mode, then TMM=0
 
-            * *side_1_both* (``int``) --
-              Direction of BOTH position, 1 as long position, -1 as short position
-              Derived from is_short
+        :param unrealized_pnl_ex_1: Unrealized PNL of all other contracts, excluding Contract 1.
```

Comment:
maybe change the variable to upnl_ex_1. Also, do we have a method to obtain this value? I'm hoping that we can get a liquidation_price formula that we are able to use within freqtrade before merging formulas in, and then realizing we have no method of calculating pnl

Reply:
@samgermain I have not gone through the codebase well enough to tell you where it will come from but the current formula for liquidation price that finance provides will need this

Reply:
https://github.com/freqtrade/freqtrade/pull/5576/commits/da00b21cd50c1341dfd97b67e72257378d5ada74

Reply:
> @samgermain I have not gone through the codebase well enough to tell you where it will come from but the current formula for liquidation price that finance provides will need this

I think we'll need to calculate it, we don't have this value stored anywhere